src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,201 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import {Ownable} from "@openzeppelin/contracts-0.8/access/Ownable.sol";
+import {Address} from "@openzeppelin/contracts-0.8/utils/Address.sol";
+import {Math} from "@openzeppelin/contracts-0.8/utils/math/Math.sol";
+import {IERC721} from "@openzeppelin/contracts-0.8/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts-0.8/token/ERC1155/IERC1155.sol";
+
+contract RequirementsRules is Ownable {
+    using Address for address;
+
+    // masStake amount allowed if user has no ERC721 or ERC1155
+    uint256 public maxStakeOverall;
+
+    struct RequirementRule {
+        uint256[] ids;
+        uint256 maxAmount;
+        uint256 reqAmount;
+        uint256 index;
+    }
+
+    mapping(IERC721 => RequirementRule) public _listERC721;
+    mapping(IERC1155 => RequirementRule) public _listERC1155;
+    IERC721[] internal _listERC721Index;
+    IERC1155[] internal _listERC1155Index;
+
+    modifier checkRequirements(
+        address account,
+        uint256 amount,
+        uint256 balanceOf
+    ) {
+        uint256 maxAllowed = _returnMaxStakeAllowed(account);
```

Comment:
maxAmount is not mandatory
if no maxAmount => we just have to check the minAmount and there is no maxStake